File: app/src/main/java/com/przemyslawlusnia/vocabularycreator/funwithtests/TestClockModule.java
package com.przemyslawlusnia.vocabularycreator.funwithtests;

import com.przemyslawlusnia.vocabularycreator.core.clock.MyClock;
import com.przemyslawlusnia.vocabularycreator.core.clock.TestMyClock;
import com.przemyslawlusnia.vocabularycreator.core.di.AppScope;
import dagger.Module;
import dagger.Provides;

@Module
public class TestClockModule {

  @Provides
  @AppScope
  MyClock myClock() {
    return new TestMyClock();
  }

}

File: app/src/main/java/com/przemyslawlusnia/vocabularycreator/funwithtests/TestClockModule.kt
package com.przemyslawlusnia.vocabularycreator.funwithtests

import com.przemyslawlusnia.vocabularycreator.core.clock.MyClock
import com.przemyslawlusnia.vocabularycreator.core.clock.TestMyClock
import com.przemyslawlusnia.vocabularycreator.core.di.AppScope
import dagger.Module
import dagger.Provides

@Module
class TestClockModule { // todo or maybe use @Qualifier and put it into normal module?

    @Provides
    @AppScope
    internal fun myClock(): MyClock {
        return TestMyClock()
    }

}
